### ISSUE TITLE
Document Xcode and modularization best practices

### ISSUE THREAD

**Issue by pepicrft** [tuist/tuist#5966]
In the past, we had a page that explained uFeatures architecture which some of our users used as a reference to modularize their efforts. We should add it to our new documentation site and use this opportunity to document what we have learned over the years.

**Comment by ladislas** [comment]
Yes please! :) We are using this architecture, it works great but not everything is straightforward 😂
